test/clojure-mode-util-test.el

```diff
@@ -374,7 +374,25 @@
         "(deftest ^{:a 1} simple-metadata)
          (deftest ^{:a {}} complex-metadata)
          (deftest |no-metadata)"
-        (expect (clojure-find-def) :to-equal '("deftest" "no-metadata")))))
+        (expect (clojure-find-def) :to-equal '("deftest" "no-metadata"))))
+  (it "should recognize defn-"
```

Comment:
Probably you should also add a couple of tests showing that anything like `defxyz-` will be accepted, so it's clearer that the `-` will work for every definition form.